platform-includes/getting-started-config/javascript.solidstart.mdx

```diff
@@ -0,0 +1,102 @@
+
+If you're using Solid Router, add the `solidRouterBrowserTracingIntegration` to collect meaningful performance data about the health of your page loads and associated requests.
+
+<SignInNote />
+
+```jsx {filename:entry-client.tsx}
+import * as Sentry from '@sentry/solidstart';
+import { solidRouterBrowserTracingIntegration } from '@sentry/solidstart/solidrouter';
+import { mount, StartClient } from '@solidjs/start/client';
+
+Sentry.init({
+  dsn: '__PUBLIC_DSN__',
+  integrations: [solidRouterBrowserTracingIntegration()],
+  tracesSampleRate: 1.0, //  Capture 100% of the transactions
+});
+
+mount(() => <StartClient />, document.getElementById('app'));
+```
+
+### Server-side Setup
+
+Create an instrument file `instrument.server.mjs` and initialize the Sentry SDK.
+
+<SignInNote />
+
+```javascript {filename:instrument.server.mjs}
+import * as Sentry from '@sentry/solidstart';
+
+Sentry.init({
+  dsn: '__PUBLIC_DSN__',
+  tracesSampleRate: 1.0, //  Capture 100% of the transactions
+});
+```
+
+### Instrumentation
+
+Complete the setup by adding the Sentry middleware to your `middleware.ts` file. If you don't have a `middleware.ts` file yet, create one:
+
+```typescript {filename:middleware.ts}
+import { sentryBeforeResponseMiddleware } from '@sentry/solidstart/middleware';
+import { createMiddleware } from '@solidjs/start/middleware';
+
+export default createMiddleware({
+  onBeforeResponse: [
+    sentryBeforeResponseMiddleware(),
+    // Add your other middleware handlers after `sentryBeforeResponseMiddleware`
+  ],
+});
+```
+
+And specify `middleware.ts` in `app.config.ts`:
+
+```typescript {filename:app.config.ts}
+import { defineConfig } from '@solidjs/start/config';
+
+export default defineConfig({
+  // ...
+  middleware: './src/middleware.ts',
+});
+```
+
+The Sentry middleware enhances the data collected by Sentry on the server side by enabling distributed tracing between the client and server.
+
+If you previously added the `solidRouterBrowserTracingIntegration` integration in `entry-client.tsx`, wrap your Solid Router with `withSentryRouterRouting`.
+This creates a higher order component, which will enable Sentry to collect navigation spans.
+
+```tsx {filename:app.tsx}
+import { Router } from "@solidjs/router";
+import { FileRoutes } from "@solidjs/start/router";
+import { withSentryRouterRouting } from '@sentry/solidstart/solidrouter'
+
+const SentryRouter = withSentryRouterRouting(Router)
+
+export default function App() {
+  return (
+    <SentryRouter>
+      <FileRoutes />
+    </SentryRouter>
+  );
+}
+```
+
+### Run your application
+
+Add an `--import` flag to the `NODE_OPTIONS` environment variable wherever you run your application.
+
+```bash {tabTitle: npm}
+NODE_OPTIONS='--import=./src/instrument.server.mjs npm start
+```
+
+```bash {tabTitle: yarn}
+NODE_OPTIONS='--import=./src/instrument.server.mjs yarn start
+```
+
+```bash {tabTitle: pnpm}
+NODE_OPTIONS='--import=./src/instrument.server.mjs pnpm start
+```
```

Comment:
L: Just a suggestion as I did this differently in the Nuxt docs ([see here](https://github.com/getsentry/sentry-docs/blob/584839eef110cc1ffbfb4f6216be1238caf89c36/platform-includes/getting-started-config/javascript.nuxt.mdx?plain=1#L61)): I added the code of how it should look like in the `package.json`. It might be easier to understand for users where they have to add it. I also included a version for when the app is started with `node` (where you can directly add the `--import` option without the env variable)

Reply:
That's a nice approach, will change!

Reply:
Updated but I opted not to use `node` directly as it's not mentioned by SolidStart nor vinxi.